.github/copilot-instructions.md

```diff
@@ -0,0 +1,24 @@
+QCoDeS is a python library that follows all the best practices of a python package.
+
+All configuration for the package and for the tools we use lives in ``pyproject.toml``.
+
+Documentation, in the ``docs/`` folder, is written using Restructured Text or Jupyter
+Notebook formats, and sphinx is used to render the documentation into HTML pages to
+be deployed at https://microsoft.github.io/Qcodes/.
+
+The documentation includes Contributor's guide, so please follow that closely.
+
+Before committing anything, run pre-commit hooks with ```pre-commit run --all`` and
+make sure they pass, and fix anything that is failing. The pre-commit hooks will ensure
+correct formatting of the code and linting as well.
+
+QCoDeS is a typed package, hence all new code should include clear and correct type annotations.
+
+We use ``pyright`` to statically check the correctness of the code with type annotations.
+Run it via ``pyright src/``.
```

Comment:
```suggestion
Run it via ``pyright``.
```
The code that should be typecheked is configured in pyproject.toml

Reply:
fixed